wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -640,21 +639,20 @@ class KotlinGenerator private constructor(
       addStatement("var %N = super.hashCode", resultName)
       beginControlFlow("if (%N == 0)", resultName)
 
-      val hashCode = MemberName("kotlin", "hashCode")
       addStatement("%N = unknownFields.hashCode()", resultName)
       for (field in type.fieldsAndFlatOneOfFields()) {
         val fieldName = localNameAllocator[field]
         add("%1N = %1N * 37 + ", resultName)
         if (field.isRepeated || field.isRequired || field.isMap) {
           addStatement("%L.hashCode()", fieldName)
         } else {
-          addStatement("%L.%M()", fieldName, hashCode)
+          addStatement("(%L?.hashCode() ?: 0)", fieldName)
```

Comment:
If this is all `kotlin.hashCode` is doing then I totally agree with inlining! FWIW I'm planning to take a look at the KotlinPoet bug and find a solution.

Reply:
Yep that's all

```kotlin
package kotlin

import kotlin.internal.InlineOnly


/**
 * Returns a hash code value for the object or zero if the object is `null`.
 *
 * @see Any.hashCode
 */
@SinceKotlin("1.3")
@InlineOnly
public inline fun Any?.hashCode(): Int = this?.hashCode() ?: 0
```